packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -1113,6 +1113,24 @@ class TextTreeRenderer {
     if (kReleaseMode) {
```

Comment:
(here and elsewhere): Wrapping things in asserts should turn the functionality off in profile _and_ release mode. Any reason we still need to keep these kReleaseMode guards around?

Reply:
Maybe performance since assert with closure alloc etc may be a little slower, don't know. Beyond the scope of this PR to change all kRelease usages across code base. I have reverted TextTreeRenderer case.

Reply:
My comment was only meant for the changes introduced in this PR. The following pattern introduced by this PR seems odd:

```dart
if (kReleaseMode) {
  return null;
}
Foo result;
assert(() {
  // Do something in debug mode.
});
return result;
```

My understanding is that in this setup the `kReleaseMode` guard is just unnecessary clutter in the code now because asserts are also disabled in release mode. With and without that `if(kReleaseMode)` the code should behave exactly the same so there shouldn't be any reason to have this extra `if(kReleaseMode)` guard.